src/main/java/org/radarcns/redcap/integration/Integrator.java

```diff
@@ -17,154 +17,50 @@
-        super(trigger, client);
-        this.mpClient = mpClient;
+    public Integrator(RedCapTrigger trigger, MpClient mpClient) {
+        this.trigger = trigger;
+        this.mpIntegrator = new MpIntegrator(mpClient);
+        this.redCapInfo = RedCapManager.getInfo(trigger);
+        this.redCapIntegrator = new RedCapIntegator(new RedCapClient(redCapInfo));
     }
 
-    /**
-     * Generates the {@link Set} of inputs that will be written in REDCap for finalising the
-     *      integration between REDCap project and Management Portal project. Using a
-     *      {@link MpClient}, the function retrieves the RADAR Subject Identifier and the Human
-     *      Readable Identifier. In the end, the function forces the REDCap integration
-     *      form /instrument status to {@link InstrumentStatus#COMPLETE}.
-     * @return {@link Set} of inputs that have to be written in REDCap.
-     */
-    @Override
-    public Set<RedCapInput> getInput() {
-        Subject subject = performSubjectUpdateOnMp(redCapInfo.getUrl(),
-                redCapInfo.getProjectId(), getRecordId());
-
-        Set<RedCapInput> set = new HashSet<>();
-
-        set.add(new IntegrationData(getRecordId(), redCapInfo.getEnrolmentEvent(),
-                IntegrationData.SUBJECT_ID_LABEL, subject.getSubjectId()));
-
-        set.add(new IntegrationData(getRecordId(), redCapInfo.getEnrolmentEvent(),
-                IntegrationData.HUMAN_READABLE_ID_LABEL, subject.getHumanReadableIdentifier()));
-
-        set.add(new IntegrationData(getRecordId(), redCapInfo.getEnrolmentEvent(),
-                RedCapManager.getStatusField(redCapInfo.getIntegrationForm()),
-                Integer.toString(InstrumentStatus.COMPLETE.getStatus())));
-
-        return set;
-    }
-
-    /**
-     * Sets the HTML form parameters needed to update the REDCap integration form / instrument.
-     *      It sets just the metadata not the real input, that is set by {@link RedCapUpdater}
-     *      while generating the {@link Request}.
-     * @param builder {@link okhttp3.FormBody.Builder} that has to be populated
-     * @return {@link okhttp3.FormBody.Builder} required to finale the HTTP request.
-     */
-    @Override
-    protected Builder setParameter(Builder builder) {
-        return builder.add("content", "record")
-                .add("format", "json")
-                .add("type", "eav")
-                .add("overwriteBehavior", "overwrite")
-                .add("returnContent", "count")
-                .add("returnFormat", "json");
+    public boolean handleDataEntryTrigger(){
+        Integer recordId = trigger.getRecord();
+        Integer projectId = redCapInfo.getProjectId();
+        String enrolmentEvent = redCapInfo.getEnrolmentEvent();
+        String integrationFrom = redCapInfo.getIntegrationForm();
+        URL url = redCapInfo.getUrl();
+        List<String> attributeKeys = redCapInfo.getAttributes().stream().map(a->a.getFieldName()).collect(Collectors.toList());
```

Comment:
attributes here can be null, make sure they are not null before starting a stream.